cmd/kube-proxy/app/server_linux.go

```diff
@@ -122,11 +124,17 @@ func (s *ProxyServer) platformCheckSupported(ctx context.Context) (ipv4Supported
 			logger.Info("No iptables support for family", "ipFamily", v1.IPv6Protocol)
 		}
 	} else {
-		// Assume support for both families.
-		// FIXME: figure out how to check for kernel IPv6 support using nft
+		// The nft CLI always supports both families.
 		ipv4Supported, ipv6Supported = true, true
 	}
 
+	// Check if the OS has IPv6 enabled, by verifying if the IPv6 interfaces are available
+	_, errIPv6 := os.Stat("/proc/net/if_inet6")
+	if errIPv6 != nil {
+		ipv6Supported = false
+	}
+	err = errors.Join(err, errIPv6)
```

Comment:
You don't want to merge `errIPv6` into `err`; returning an error means "kube-proxy can't run at all".

Oh, but also, for consistency with the iptables userspace tests, if the ipv6 check fails, can you add:

```
		logger.Info("No kernel support for family", "ipFamily", v1.IPv6Protocol)
```

Reply:
fixed and squashed.